pkg/crawl/identify.go

```diff
@@ -189,15 +202,46 @@ func lookForMatchInFile(ctx context.Context, path string, size int64, contents i
 		return JarNameInsideArchive, version, nil
 	}
 	if strings.HasSuffix(path, "JndiManager.class") {
-		version, md5Match := classMd5Version(contents)
-		if md5Match {
-			return ClassName | ClassFileMd5, version, nil
+		finding, version, hashMatch := lookForHashMatch(contents)
+		if hashMatch {
+			return ClassName | finding, version, nil
 		}
 		return ClassName, parentVersion, nil
 	}
 	return NothingDetected, parentVersion, nil
 }
 
+const maxClassSize = 0xffff
+
+var classByteBuf []byte = make([]byte, maxClassSize)
+
+func lookForHashMatch(contents io.Reader) (Finding, string, bool) {
+	i := 0
+	for {
+		read, err := contents.Read(classByteBuf[i:])
+		i += read
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return NothingDetected, UnknownVersion, false
+		}
+		if read > maxClassSize {
+			return NothingDetected, UnknownVersion, false
+		}
+	}
```

Comment:
You can use `bytes.Buffer` which will do this for you, and then use `buff.Bytes()` to get the byte array
```suggestion
	buff := new(bytes.Buffer)
	if _, err := buff.ReadFrom(contents); err != nil {
		return NothingDetected, UnknownVersion, false
	}
```

Reply:
The current more manual way was intentional for a couple of reasons:
- Wanting to limit maximum we could read to more than a log4j class but smaller than what could appear
- Reusing the same buffer across multiple calls to avoid as much gc churn as possible, but this might have been a premature optimisation. I wouldn't have done it for this alone but with the first point it seemed easy.
If there's a good way of doing this with Buffer or similar I'm happy to swap over, the important point is limiting how much will be read in to memory.

Reply:
I was reading the docs wrong and buffer is better than my code. Updating.